src/Components/Components/src/Binding/CascadingModelBinder.cs

```diff
@@ -0,0 +1,67 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System.Reflection.Metadata;
+using Microsoft.AspNetCore.Components.Binding;
+
+namespace Microsoft.AspNetCore.Components;
+
+/// <summary>
+/// Defines the binding context for data bound from external sources.
+/// </summary>
+public class CascadingModelBinder : IComponent
+{
+    private RenderHandle _handle;
+    private bool _hasRendered;
+    private ModelBindingContext? _bindingContext;
+
+    /// <summary>
+    /// The binding context name.
+    /// </summary>
+    [Parameter] public string Name { get; set; } = default!;
+
+    /// <summary>
+    /// The binding context name.
+    /// </summary>
+    [Parameter] public string BindingContextId { get; set; } = default!;
+
+    /// <summary>
+    /// Specifies the content to be rendered inside this <see cref="CascadingModelBinder"/>.
+    /// </summary>
+    [Parameter] public RenderFragment<ModelBindingContext> ChildContent { get; set; } = default!;
+
+    [CascadingParameter] ModelBindingContext? ParentContext { get; set; }
+
+    void IComponent.Attach(RenderHandle renderHandle)
+    {
+        _handle = renderHandle;
+    }
+
+    Task IComponent.SetParametersAsync(ParameterView parameters)
+    {
+        if (!_hasRendered)
```

Comment:
I'm not sure this logic will work. If you had a structure like this:

```razor
@page "/mypage"

<CascadingModelBinder>
    <p>The count is @count</p>
</CascadingModelBinder>

<button @onclick="IncrementCounter">Click me</button>
```

... will it behave as expected? I'm suspicious that it will never update the count UI because it's not re-rendering the child content fragment.

Reply:
Yeah, I've updated the logic here